src/transaction.js

```diff
@@ -1221,21 +1223,24 @@ Transaction.prototype.summary = function() {
     initialSubmitIndex: this.initialSubmitIndex,
     lastLedgerSequence: this.lastLedgerSequence,
     state: this.state,
-    server: this._server ? this._server._opts.url : undefined,
     finalized: this.finalized
   };
 
   if (this.result) {
-    result.result = {
+    txSummary.result = {
       engine_result: this.result.engine_result,
       engine_result_message: this.result.engine_result_message,
       ledger_hash: this.result.ledger_hash,
       ledger_index: this.result.ledger_index,
-      transaction_hash: this.result.tx_json.hash
+      transaction_hash: undefined
     };
+
+    if (this.result.tx_json) {
+      txSummary.result.transaction_hash = this.result.tx_json.hash;
```

Comment:
Could avoid mutation by declaring a variable for `transaction_hash` before `txSummary.result = ...`, or use a ternary expression in the `txSummary.result = ...` assignment.